### PR TITLE
Update core to latest & fix impacted components

### PR DESCRIPTION
**Pull request by gramidt**
**Description:**
Updated core and updated impacted components with the latest Client Authentication Extension addition. 

Impacted components:

Exporters:

- lokiexporter (I'm the codeowner)
- dynatraceexporter
- sumologicexporter
- influxdbexporter
- signalfxexporter
- f5cloudexporter (I'm the codeowner)
- awsprometheusremotewriteexporter
- humioexporter
- uptraceexporter

Receivers:
- nginxreceiver

Extensions:
- httpforwarderextension

I have split up each component into their own commit and ensured all changes to that component remained in their corresponding commit.

**Link to tracking Issue:**
#3438 

**Testing:**
Unit tests

**Documentation:**
Unit tests

**Notes**
This one was a fun one. PLEASE TAKE EXTRA CARE AND ASSUME I MISSED SOMETHING when reviewing your components in this PR, as some were straightforward others required more effort.

### CODE REVIEW

**Comment by gramidt** [comment]
The first round of the build and test errors have been sorted out. I am now reviewing all components that use the confighelpers to see if there's a clash in settings ( "auth" field). I know 100% this is an issue with the f5cloudexporter.

**Comment by gramidt** [comment]
Confirmed list of components that use the impacted confighelpers and have a config setting field named "auth" at the top level:

- f5cloudauth (Codeowner: gramidt)

**Comment by gramidt** [comment]
Since I am the codeowner of the `f5cloudexporter` I will update it in this PR to avoid issues.

**Comment by gramidt** [comment]
For some reason the humioexporter didn't show up in earlier failures. Working on it now...

**Comment by gramidt** [comment]
Now working on the uptrace exporter...

**Comment by gramidt** [comment]
This one was a fun one. Please take EXTRA CARE when reviewing your components in this PR, as some were straightforward others required more effort. 

I have split up each component into their own commit and ensured all changes to that component remained in their corresponding commit.

**Comment by gramidt** [comment]
cc @tigrannajaryan @jpkrohling @pavankrish123

**Comment by gramidt** [comment]
There appears to be a flaky test for the signalfx exporter. I will dig into this ASAP.

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8d1814]

goroutine 19 [running]:
net/http.(*Client).deadline(0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:189 +0x34
net/http.(*Client).do(0x0, 0xc00022a500, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:604 +0xbd
net/http.(*Client).Do(...)
	/usr/local/go/src/net/http/client.go:586
github.com/signalfx/signalfx-agent/pkg/apm/requests.sendRequest(0x0, 0xc00022a500, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/go/pkg/mod/github.com/signalfx/signalfx-agent/pkg/apm@v0.0.0-20201202163743-65b4fa925fc8/requests/sender.go:116 +0x8f
github.com/signalfx/signalfx-agent/pkg/apm/requests.(*ReqSender).sendRequest(0xc0000ce780, 0xc00022a500, 0x1, 0x1)
	/home/circleci/go/pkg/mod/github.com/signalfx/signalfx-agent/pkg/apm@v0.0.0-20201202163743-65b4fa925fc8/requests/sender.go:72 +0x79
github.com/signalfx/signalfx-agent/pkg/apm/requests.(*ReqSender).processRequests(0xc0000ce780)
	/home/circleci/go/pkg/mod/github.com/signalfx/signalfx-agent/pkg/apm@v0.0.0-20201202163743-65b4fa925fc8/requests/sender.go:62 +0x256
created by github.com/signalfx/signalfx-agent/pkg/apm/requests.(*ReqSender).Send
	/home/circleci/go/pkg/mod/github.com/signalfx/signalfx-agent/pkg/apm@v0.0.0-20201202163743-65b4fa925fc8/requests/sender.go:44 +0x11a
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/correlation	0.047s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/dimensions	20.081s	coverage: 85.4% of statements
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/hostmetadata	0.089s	coverage: 100.0% of statements
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/translation	0.189s	coverage: 96.9% of statements
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter/translation/dpfilters	0.056s	coverage: 100.0% of statements
FAIL
```

**Comment by tigrannajaryan** [comment]
> There appears to be a flaky test for the signalfx exporter. It has failed twice now and has passed other times. I will dig into this ASAP.

Please file a bug so that we fix the flaky test.

**Review by tigrannajaryan — COMMENTED** [review]
@gramidt thank you for working on this.

File: exporter/awsprometheusremotewriteexporter/auth_test.go
Comment (jpkrohling) [inline exporter/awsprometheusremotewriteexporter/auth_test.go:113]: If you prefer, you can use something like `map[config.ComponentID]component.Extension{}` instead.

File: exporter/f5cloudexporter/README.md
Comment (jpkrohling) [inline exporter/f5cloudexporter/README.md:34]: Just out of curiosity: are you converting this into an authenticator in the future?

**Review by jpkrohling — APPROVED** [review]
I'm approving the components that I own and the general semantics of the auth changes. Other component owners still need to give their approvals.

File: exporter/f5cloudexporter/README.md
Comment (gramidt) [reply exporter/f5cloudexporter/README.md:34]: Yeah, that the intention. I will be working on an F5 Authenticator as soon as this is merged and will likely be deprecating this exporter in favor of the OTLP gRPC + F5 Auth Ext :)

File: exporter/awsprometheusremotewriteexporter/auth_test.go
Comment (gramidt) [reply exporter/awsprometheusremotewriteexporter/auth_test.go:113]: That's also a great option, @jpkrohling! Which way do you believe is the correct convention for us to use moving forward? I know we typically use `componentest` when possible, so I thought following that pattern made the most sense at the time of this PR.

**Comment by gramidt** [comment]
> @gramidt thank you for working on this.

Anytime, @tigrannajaryan! It's all about the community :)

**Comment by gramidt** [comment]
It seems that "Publish-2" had an issue. I'll re-kick this off when we're ready.

https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/14123/workflows/48ca9007-bc9e-4aac-a634-4626559eaeec/jobs/121072

**Comment by gramidt** [comment]
> > There appears to be a flaky test for the signalfx exporter. It has failed twice now and has passed other times. I will dig into this ASAP.
> 
> Please file a bug so that we fix the flaky test.

I believe it was caused by me and I fixed it. I have been running the entire suite locally without seeing that issue again. Once everyone has had the opportunity to review, I will re-kick the tests a few times to be sure. If I see it again, I will file an issue (assuming I didn't cause it with these changes).

**Comment by pavankrish123** [comment]
Thank you @gramidt

**Comment by gramidt** [comment]
Just rebased to get latest changes.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by gramidt** [comment]
Friendly ping CODEOWNERS.

This PR is critical since it's needed before we can pull any further updates from the core collector.

File: exporter/dynatraceexporter/metrics_exporter.go
Comment (bogdandrutu) [inline exporter/dynatraceexporter/metrics_exporter.go:231]: This is not used in the factory as "WithStart" so it will not be called.

File: exporter/humioexporter/traces_exporter.go
Comment (bogdandrutu) [inline exporter/humioexporter/traces_exporter.go:262]: Same, This is not used in the factory as "WithStart" so it will not be called.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
There are too many changes that break the components. We need to rollback the change in core, and make PRs to move the http client initialization in Start as a separate PR.

**Comment by gramidt** [comment]
@jpkrohling @pavankrish123 - Please see @bogdandrutu comments above. Sadly, I over extended myself to work on this PR to help out, but I do not have the necessary bandwidth to rollback and adjust this effort in the near future. Would either of you be able to work with @bogdandrutu on this?

**Comment by bogdandrutu** [comment]
@gramidt I think you can start splitting from this PR, smaller PRs that simply change where ToClient is called (in start). Then when the change to the API is rolledback in core we can simply just add the extensions to the call.

**Comment by gramidt** [comment]
I should be able to manage that, @bogdandrutu! I will start now and see how far I get with my limited time today.

**Comment by gramidt** [comment]
Closing this in favor of @bogdandrutu approach.

**Pull request closed without merging**
